packages/@lwc/engine-core/src/renderer.ts

```diff
@@ -10,267 +13,75 @@ export type HostElement = any;
-    getElementsByClassName = getElementsByClassNameImpl;
-}
-
-type getChildrenFunc = (element: E) => HTMLCollection;
-export let getChildren: getChildrenFunc;
-export function setGetChildren(getChildrenImpl: getChildrenFunc) {
-    getChildren = getChildrenImpl;
-}
-
-type getChildNodesFunc = (element: E) => NodeList;
-export let getChildNodes: getChildNodesFunc;
-export function setGetChildNodes(getChildNodesImpl: getChildNodesFunc) {
-    getChildNodes = getChildNodesImpl;
-}
-
-type getFirstChildFunc = (element: E) => N | null;
-export let getFirstChild: getFirstChildFunc;
-export function setGetFirstChild(getFirstChildImpl: getFirstChildFunc) {
-    getFirstChild = getFirstChildImpl;
-}
-
-type getFirstElementChildFunc = (element: E) => E | null;
-export let getFirstElementChild: getFirstElementChildFunc;
-export function setGetFirstElementChild(getFirstElementChildImpl: getFirstElementChildFunc) {
-    getFirstElementChild = getFirstElementChildImpl;
-}
-
-type getLastChildFunc = (element: E) => N | null;
-export let getLastChild: getLastChildFunc;
-export function setGetLastChild(getLastChildImpl: getLastChildFunc) {
-    getLastChild = getLastChildImpl;
-}
-
-type getLastElementChildFunc = (element: E) => E | null;
-export let getLastElementChild: getLastElementChildFunc;
-export function setGetLastElementChild(getLastElementChildImpl: getLastElementChildFunc) {
-    getLastElementChild = getLastElementChildImpl;
-}
-
-type isConnectedFunc = (node: N) => boolean;
-export let isConnected: isConnectedFunc;
-export function setIsConnected(isConnectedImpl: isConnectedFunc) {
-    isConnected = isConnectedImpl;
-}
-type insertStylesheetFunc = (content: string, target?: ShadowRoot) => void;
-export let insertStylesheet: insertStylesheetFunc;
-export function setInsertStylesheet(insertStylesheetImpl: insertStylesheetFunc) {
-    insertStylesheet = insertStylesheetImpl;
-}
-
-type assertInstanceOfHTMLElementFunc = (elm: any, msg: string) => void;
-export let assertInstanceOfHTMLElement: assertInstanceOfHTMLElementFunc;
-export function setAssertInstanceOfHTMLElement(
-    assertInstanceOfHTMLElementImpl: assertInstanceOfHTMLElementFunc
-) {
-    assertInstanceOfHTMLElement = assertInstanceOfHTMLElementImpl;
-}
-
-type defineCustomElementFunc = (
-    name: string,
-    constructor: CustomElementConstructor,
-    options?: ElementDefinitionOptions
-) => void;
-export let defineCustomElement: defineCustomElementFunc;
-export function setDefineCustomElement(defineCustomElementImpl: defineCustomElementFunc) {
-    defineCustomElement = defineCustomElementImpl;
-}
-
-type getCustomElementFunc = (name: string) => CustomElementConstructor | undefined;
-export let getCustomElement: getCustomElementFunc;
-export function setGetCustomElement(getCustomElementImpl: getCustomElementFunc) {
-    getCustomElement = getCustomElementImpl;
+let defaultRenderer: RendererAPI;
```

Comment:
What is the value of keeping around `defaultRender`? Since all the VM have an associated renderer, the renderer can be passed as an argument to the patching method: `patchChildren` and `hydrateChildren`.

Reply:
I will like to explore that. You still need to have access to the defaultRenderer though

Reply:
done something similar, instead of default renderer, I'm passing the proper renderer, whether it is default or not is relevant... take another look.

Reply:
Nice.

Now speaking about the `getRendererFromVNode`. If we put aside all the usage of `modules/*.ts`, this function is only used in `patch` and `mount`. In both places, we have access to the owner's renderer. We could update the `getRendererFromVNode` renderer to completely get rid of this concept of default renderer:

```ts
export function getRendererFromVNode(vnode: VBaseElement, fallbackRenderer: RendererAPI): RendererAPI {
    return vnode.data.renderer ?? fallbackRenderer;
}
```

Reply:
it is gone!